### PR TITLE
Update section about exceptions

### PR DESCRIPTION
**Pull request by melnikovi**
## Purpose of this pull request

Update section about exceptions in technical guidelines.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html

## Links to Magento source code

- ...

whatsnew
Added a [technical guideline](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html) 5.20 about exceptions in observers and plugins.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=5209) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=5209) before we can accept your contribution.<br/>

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/coding-standards/technical-guidelines.md [hunk 1]
```diff
@@ -482,6 +482,8 @@ You need to read configuration from different sources (like database or filesyst
 
 5.19. Each module or component MUST declare its own exceptions. Exceptions declared in other components SHOULD NOT be thrown.
 
+5.20 Observers and plugins MUST throw only exceptions that can be thrown by the method where event is triggered or by the method that is being pluginized.
```
Comment (buskamuza) [inline guides/v2.2/coding-standards/technical-guidelines.md]:
```suggestion
5.20 Observers and plugins MUST throw only exceptions that can be thrown by the method where the event is triggered or by the method that is being pluginized.
```

File: guides/v2.2/coding-standards/technical-guidelines.md [hunk 2]
```diff
@@ -482,6 +482,8 @@ You need to read configuration from different sources (like database or filesyst
 
 5.19. Each module or component MUST declare its own exceptions. Exceptions declared in other components SHOULD NOT be thrown.
 
+5.20. Plugin MUST throw only exceptions declared by the method to which the plugin is added. Observer MUST throw only exceptions declared by the method that triggers an event.
```
Comment (erikmarr) [inline guides/v2.2/coding-standards/technical-guidelines.md]:
```suggestion
5.20. Plugin MUST only throw exceptions declared by the method to which the plugin is added. Observer MUST only throw exceptions declared by the method that triggers an event.
```

**Comment by jeff-matthews** [comment]
running tests

**Comment by dshevtsov** [comment]
@jeff-matthews this pull request still require a review form an architect @maghamed .

**Comment by jeff-matthews** [comment]
Thanks @dshevtsov. I forgot that _all_ architect reviewers must approve; not just one. 

Waiting for your approval @maghamed.

**Comment by dshevtsov** [comment]
running tests

**Comment by dshevtsov** [comment]
running tests

**Comment by dshevtsov** [comment]
running tests

**Pull request merged**